src/codewhisperer/commands/basicCommands.ts

```diff
@@ -43,14 +57,17 @@ export const enableCodeSuggestions = Commands.declare(
         await vscode.commands.executeCommand('setContext', 'CODEWHISPERER_ENABLED', true)
         await vscode.commands.executeCommand('aws.codeWhisperer.refresh')
 
-        const hasShownWelcomeMsgBefore = get(
-            CodeWhispererConstants.welcomeMessageKey,
+        const hasShownWelcomeMsgBeforeToExistingUsers = get(
+            CodeWhispererConstants.welcomeMessageKeyShown,
             context.extensionContext.globalState
```

Comment:
if [PromptSettings](https://github.com/aws/aws-toolkit-vscode/pull/3805/files#r1336497017) is used then this would become `settings.isPromptEnabled('...')`